verifier/src/channel.rs

```diff
@@ -177,9 +188,34 @@ impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> VerifierChanne
         let queries = self.trace_queries.take().expect("already read");
 
         // make sure the states included in the proof correspond to the trace commitment
-        for (root, proof) in self.trace_roots.iter().zip(queries.query_proofs.iter()) {
-            MerkleTree::verify_batch(root, positions, proof)
-                .map_err(|_| VerifierError::TraceQueryDoesNotMatchCommitment)?;
+
+        let items: Vec<H::Digest> =
+            { queries.main_states.rows().map(|row| H::hash_elements(row)).collect() };
```

Comment:
I'm not sure curly brackets are really needed here.

Reply:
removed